components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -1729,6 +1729,7 @@ private IdentityProvider createIDP(IdentityProviderPOSTRequest identityProviderP
         IdentityProvider idp = new IdentityProvider();
         idp.setIdentityProviderName(identityProviderPOSTRequest.getName());
         idp.setAlias(identityProviderPOSTRequest.getAlias());
+        idp.setTemplateId(identityProviderPOSTRequest.getTemplateId());
```

Comment:
is the null/empty cases handled properly?

Reply:
Validates before adding to the DB. https://github.com/AnuradhaSK/carbon-identity-framework/blob/f30b880341f0b85c4fa88a9959d53a86865b1e23/components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java#L2037-L2045